WalletWasabi/WabiSabi/Models/MultipartyTransaction/SigningState.cs

```diff
@@ -22,13 +24,17 @@ public SigningState(MultipartyTransactionParameters parameters, IEnumerable<Coin
 		public bool IsFullySigned => Witnesses.Count == Inputs.Count;
 
 		[JsonIgnore]
-		public IEnumerable<Coin> UnsignedInputs => Inputs.Where((_, i) => !IsInputSigned(i));
+		public IEnumerable<Coin> UnsignedInputs => GetSortedInputs().Where((_, i) => !IsInputSigned(i));
 
 		public bool IsInputSigned(int index) => Witnesses.ContainsKey(index);
 
 		public bool IsInputSigned(OutPoint prevout) => IsInputSigned(GetInputIndex(prevout));
 
-		public int GetInputIndex(OutPoint prevout) => Inputs.ToList().FindIndex(coin => coin.Outpoint == prevout); // this is inefficient but is only used in tests, see also dotnet/runtime#45366
+		public int GetInputIndex(OutPoint prevout) =>
+			GetSortedInputs()
+			.Select((x,i) => (OutPoint: x.Outpoint, Index: i))
+			.First(x => x.OutPoint == prevout)
+			.Index; // this is inefficient but is only used in tests, see also dotnet/runtime#45366
```

Comment:
The comment applies to the computed property and the use of `First()`

```suggestion
		 // this is inefficient but is only used in tests, see also dotnet/runtime#45366
		 public int GetInputIndex(OutPoint prevout) =>
			GetSortedInputs()
			.Select((x,i) => (OutPoint: x.Outpoint, Index: i))
			.First(x => x.OutPoint == prevout)
			.Index;
```

Reply:
Restored